src/mm/device_alloc.rs

```diff
@@ -37,8 +29,6 @@ unsafe impl Allocator for DeviceAlloc {
 		let addr = ptr.as_ptr().expose_provenance().into();
 
 		if let Some(phys_addr) = arch::mm::paging::virtual_to_physical(addr) {
```

Comment:
This is now redundant, virtual and physical address are the same, we can just directly deallocate the address returned by `expose_provenance`.

Reply:
Yes! Thanks! :)